Clickhouse `file` table function parses incorrectly and can't round-trip

**Before you file an issue**
- Make sure you specify the "read" dialect eg. `parse_one(sql, read="spark")` Clickhouse
- Make sure you specify the "write" dialect eg. `ast.sql(dialect="duckdb")` Clickhouse
- Check if the issue still exists on main

**Fully reproducible code snippet**
Please include a fully reproducible code snippet or the input sql, dialect, and expected output.

```python
sqlglot.parse_one("CREATE VIEW myschema.myview (c1 String NOT NULL) AS SELECT c1  FROM file('base/dir/*.parquet', Parquet)", dialect="clickhouse").sql(dialect="clickhouse")

```

Returns this:

```sql
CREATE VIEW myschema.myview (c1 String NOT NULL) AS SELECT c1 FROM file AS _t0("base/dir/*.parquet", Parquet)
```

Note the `AS _t0`

Interestingly the [s3 table function](https://clickhouse.com/docs/sql-reference/table-functions/s3) is understood.

```python
sqlglot.parse_one("CREATE VIEW myschema.myview (c1 String NOT NULL) AS SELECT c1  FROM s3('base/dir/*.parquet', Parquet)", dialect="clickhouse").sql(dialect="clickhouse")

```
return this:
```sql
CREATE VIEW myschema.myview (c1 String NOT NULL) AS SELECT c1 FROM s3('base/dir/*.parquet', Parquet)
```

I came across this while to create DDL for clickhouse views that read parquet files. For now, my workaround is something like this:

```python
    from_clause = exp.From(
        this=exp.Table(
            this=exp.Anonymous(
                this="file",
                expressions=[
                    exp.Literal(this=f"{ctx.file_path}/*.parquet", is_string=True),
                    exp.Column(this=exp.Identifier(this="Parquet", quoted=False)),
                ],
            )
        )
    )
```

## ROOT CAUSE

The AST (remove `.sql(dialect="clickhouse")` at the end is the following) looks like this for the `s3` table function vs the `file` table function

INCORRECT WITH `file` FUNCTION::
```python
From(
      this=Table(
        this=Identifier(this=file, quoted=False),
        alias=TableAlias(
          columns=[
            Identifier(this='base/dir/*.parquet', quoted=True),
            Identifier(this=Parquet, quoted=False)])))
``` 

CORRECT WITH `s3` TABLE FUNCTION
```python
From(
      this=Table(
        this=Anonymous(
          this=s3,
          expressions=[
            Literal(this='base/dir/*.parquet', is_string=True),
            Column(
              this=Identifier(this=Parquet, quoted=False))])))
```
Note `TableAlias` vs `Anonymous`.

I had some trouble finding out if these are custom-registered functions, but couldn't find anything clickstore specific. 

```
$ rg "s3" sqlglot/
sqlglot/generators/doris.py
459:        "s3",

sqlglot/generators/singlestore.py
1167:        "s3",
```
vs
```
$ rg '"file"' sqlglot/
sqlglot/generators/exasol.py
576:        "file",

sqlglot/generators/singlestore.py
689:        "file",

sqlglot/generators/doris.py
252:        "file",
```

I have also made a reproducible test here:

https://github.com/varoonp123/sqlglot/commit/9afac61d14fffec819afe4cc0e613fb1aae36499

```
AssertionError: "CREATE VIEW myschema.myview (c1 String N[58 chars]uet)" != 'CREATE VIEW myschema.myview (c1 String N[65 chars]uet)'
- CREATE VIEW myschema.myview (c1 String NOT NULL) AS SELECT c1 FROM file('base/dir/*.parquet', Parquet)
?                                                                         ^                  ^
+ CREATE VIEW myschema.myview (c1 String NOT NULL) AS SELECT c1 FROM file AS _t0("base/dir/*.parquet", Parquet)

```

I didnt spend a huge amount of time looking, but nothing stuck out here: https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/clickhouse.py

Ill see if I can find some time to take a look at: https://github.com/tobymao/sqlglot/blob/1d07352dfb421dce825eb47e825c3b6124ea4bd3/sqlglot/parsers/clickhouse.py#L608

If anyone can help narrow down the search before I find time to look at this more closely, I can put up a fix.

**Official Documentation**
The Clickhouse table function [file](https://clickhouse.com/docs/sql-reference/table-functions/file) is used for reading local files.

--- a/sqlglot/parsers/clickhouse.py
+++ b/sqlglot/parsers/clickhouse.py
@@ -316,6 +316,7 @@ class ClickHouseParser(parser.Parser):
     FUNC_TOKENS = {
         *parser.Parser.FUNC_TOKENS,
         TokenType.AND,
+        TokenType.FILE,
         TokenType.OR,
         TokenType.SET,
     }

--- a/tests/dialects/test_clickhouse.py
+++ b/tests/dialects/test_clickhouse.py
@@ -1855,6 +1855,13 @@ LIFETIME(MIN 0 MAX 0)""",
         )
         self.validate_identity("splitByChar('', x)")
 
+    def test_table_functions(self):
+        self.validate_identity(
+            "CREATE VIEW myschema.myview (c1 String NOT NULL) AS SELECT c1 FROM file('base/dir/*.parquet', Parquet)"
+        )
+        self.validate_identity("SELECT * FROM file('path', Parquet) WHERE x > 1")
+        self.validate_identity("SELECT * FROM file('path', Parquet)")
+
     def test_sql_security(self):
         stmts = [
             "CREATE VIEW v DEFINER='alice' SQL SECURITY DEFINER AS SELECT 1",

